plottr/data/qcodes_dataset.py

```diff
@@ -211,6 +211,7 @@ class QCodesDSLoader(Node):
     def __init__(self, *arg: Any, **kw: Any):
         self._pathAndId: Tuple[Optional[str], Optional[int]] = (None, None)
         self.nLoadedRecords = 0
+        self._dataset: Optional[DataSet] = None
```

Comment:
is it true that at the moment there are no use-cases/usages of `QCodesDSLoader` in `plottr` where `QCodesDSLoader` does NOT get reinstantiated in order to load a different dataset? Seeing the existence of `@pathAndId.setter`, makes me think that before this change it was potentially possible to "send" new `_pathAndId` tuple to `QCodesDSLoader` node and that would load this new dataset in `process`.

perhaps the implementation of `process` could be improved to call `load_dataset_from` if `_pathAndId` "changed" (or if `_dataset` is `None`, of course) ?

Reply:
I don't think this happens in the current codebase but lets be safe. I have changed the code so that the `_dataset` attribute is cleared when `_pathAndId` changes

Reply:
awesome!